### PR TITLE
Moderation

### PR DESCRIPTION
**Pull request by JulianVennen**
Commands: mute, ban, kick, unmute, unban, userinfo
temporary bans/mutes

### CODE REVIEW

File: bot.js [hunk 1]
```diff
@@ -44,22 +38,52 @@ const database = new Database(config.db);
         }
     }
 
-    // load features
-    const features = [];
-    for (let file of await fs.readdir(`${__dirname}/features`)) {
-        let path = `${__dirname}/features/${file}`;
+    // FEATURES
+    // message
```
Comment (patricevoss) [inline bot.js:42]: Please add a more descriptive comment.

File: bot.js [hunk 2]
```diff
@@ -44,22 +38,52 @@ const database = new Database(config.db);
...
+    const messages = [];
+    for (let file of await fs.readdir(`${__dirname}/features/message`)) {
+        let path = `${__dirname}/features/message/${file}`;
+        if (!file.endsWith('.js') || !(await fs.lstat(path)).isFile()) {
```
Comment (patricevoss) [inline bot.js:46]: Please add line break

File: bot.js [hunk 3]
```diff
@@ -44,22 +38,52 @@ const database = new Database(config.db);
...
+            continue;
+        }
+        try {
+            let feature = require(path);
+            messages.push(feature);
```
Comment (patricevoss) [inline bot.js:51]: Redundant variable. Please read PhpStorm recommendations.

File: bot.js [hunk 4]
```diff
@@ -44,22 +38,52 @@ const database = new Database(config.db);
...
+        } catch (e) {
+            console.error(`Failed to load message feature '${file}'`, e);
+        }
+    }
+    // guildMemberAdd
```
Comment (patricevoss) [inline bot.js:56]: WTF is this comment

File: bot.js [hunk 5]
```diff
@@ -69,16 +93,27 @@ const database = new Database(config.db);
...
         for (let command of commands) {
             if (command.names.includes(cmd)) {
-                await Promise.resolve(command.command(message, args, channels, database));
+                await Promise.resolve(command.command(message, args, database, bot));
                 break;
             }
         }
     });
+
+    //FEATURES
```
Comment (patricevoss) [inline bot.js:102]: dude…

File: checks/servers.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  let channelIds = await database.queryAll("SELECT DISTINCT id FROM channels");
+
+  for (let channelId of channelIds) {
+    let channel = await util.getChannelConfig(channelId.id);
+    let relevant = Math.floor(Date.now() / 1000) - (isNaN(channel.cooldown) ? 0 : channel.cooldown);
+    await database.query("DELETE FROM servers WHERE channelid = ? AND timestamp <= ?", [channel.id, relevant]);
+  }
+  //stats on servers size
+  let result = await database.query("SELECT COUNT(*) as c FROM servers");
+  let date = new Date();
```
Comment (patricevoss) [inline checks/servers.js:14]: Please define variables on top of the function

File: commands/mute.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+    member.roles.add(mutedRole, `moderator: ${message.author.username}#${message.author.discriminator} duaration: ${time} reason: ` + reason);
+    let insert = await database.queryAll("INSERT INTO moderations (guildid, userid, action, created, expireTime, reason, moderator) VALUES (?,?,?,?,?,?,?)",[message.guild.id, userId, 'mute', now, endsAt, reason, message.author.id]);
+
+    message.channel.send(`Muted \`${member.user.username}#${member.user.discriminator}\` for ${time}: ${reason}`);
+    util.log(message, `\`[${insert.insertId}]\` \`${message.author.username}#${message.author.discriminator}\` muted \`${member.user.username}#${member.user.discriminator}\` for ${time}: ${reason}`);
+  }
+  else {
+    member.roles.add(mutedRole, `${message.author.username}#${message.author.discriminator}: `+reason);
+    let insert = await database.queryAll("INSERT INTO moderations (guildid, userid, action, created, reason, moderator) VALUES (?,?,?,?,?,?)",[message.guild.id, userId, 'mute', now, reason, message.author.id]);
```
Comment (patricevoss) [inline commands/mute.js:61]: ?,?,?,?,?,?

File: lib/util.js [hunk 8]
```diff
@@ -107,32 +126,98 @@ util.isTime = (word) => {
...
+  setTimeout(() => {
+    channels.delete(result.id);
+  },cacheDuration);
+  return "Success!";
+}
+
+util.isMod = async (member) => {
+  let guildConfig = await util.getGuildConfig(member.guild);
+  for (let [key,role] of member.roles.cache) {
+    if (guildConfig.isModRole(key))
```
Comment (patricevoss) [inline lib/util.js:217]: WTF is this ugly braces shit

File: commands/mute.js (hunk 7 above)
Comment (JulianVennen) [reply commands/mute.js:61]: !,!,!,!,!,!

**Pull request merged**